titanic.py

```diff
@@ -91,8 +91,8 @@ def maximum_energy_delta(bqm):
     """Compute conservative bound on maximum change in energy when flipping a single variable"""
     return max(abs(bqm.get_linear(i))
                + sum(abs(bqm.get_quadratic(i,j))
-                     for j in bqm.iter_neighbors(i))
-               for i in bqm.iter_variables())
+                     for j, _ in bqm.iter_neighborhood(i))
```

Comment:
This doesn't work in `dimod~=0.9.0` (Ocean < 4).

I don't think there's a method for iterating neighbors that exists in both dimod 0.9 and 0.10, but @arcondello should confirm.

If that's so, then we have a choice:
- lower-bound examples to Ocean 4
- tolerate deprecation messages
- de-deprecate `iter_neighbors`/`iter_quadratic` and/or rename `iter_neighborhood` to `iter_neighbors`.

I'm not sure why we deprecated `iter_quadratic`.

Reply:
This is the only example that used `iter_neighbors`, so I would vote for raising the lower bound.

Reply:
That makes sense, @hhtong, let's raise the lower bound for this example.

Reply:
Done